### PR TITLE
fix(wallet): BTC Offramp

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Fixes `Bitcoin` sell with `Ramp`

Resolves <https://github.com/brave/brave-browser/issues/35796>

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
   1. Fund your `Wallet` with some `Bitcoin`
   2. Open token details screen
   3. Select more menu for account and select `Sell`
   4. Enter value and click on `Sell on Ramp`, it should load the `Ramp Offramp` page

Before:

https://github.com/brave/brave-core/assets/40611140/6084ef03-d1c5-4c51-82e3-bf5ab96b07ce

After:

https://github.com/brave/brave-core/assets/40611140/aa10e46f-e3eb-45b5-8905-1c82431e1170

### CODE REVIEW

**Review by StephenHeaps — APPROVED** [review]
iOS ++

File: components/brave_wallet/browser/brave_wallet_constants.h [hunk 1]
```diff
@@ -30,9 +30,10 @@ constexpr int32_t kAssetDiscoveryMinutesPerRequest = 1;
 inline constexpr char kWalletBaseDirectory[] = "BraveWallet";
 inline constexpr char kImageSourceHost[] = "erc-token-images";
 inline constexpr char kRampBaseUrl[] = "https://app.ramp.network";
-inline constexpr char kOffRampEnabledFlows[] = "ONRAMP,OFFRAMP";
-inline constexpr char kOffRampDefaultFlow[] = "OFFRAMP";
-inline constexpr char kRampID[] = "8yxja8782as5essk2myz3bmh4az6gpq4nte9n2gf";
+inline constexpr char kOffRampEnabledFlows[] = "OFFRAMP";
+inline constexpr char kOnRampEnabledFlows[] = "ONRAMP";
+inline constexpr char kOnRampID[] = "8yxja8782as5essk2myz3bmh4az6gpq4nte9n2gf";
```
Comment (supermassive) [inline components/brave_wallet/browser/brave_wallet_constants.h:35]: nit: `Flow`` instead of `Flows`?

Comment (supermassive) [reply components/brave_wallet/browser/brave_wallet_constants.h:35]: Well I think Flows is also ok there

File: components/brave_wallet_ui/common/hooks/use-multi-chain-sell-assets.ts [hunk 2]
```diff
@@ -79,7 +77,6 @@ export const useMultiChainSellAssets = () => {
           assetSymbol: getRampAssetSymbol(sellAsset, true),
           offRampProvider: BraveWallet.OffRampProvider.kRamp,
           chainId: sellAsset.chainId,
-          address: sellAddress,
           amount: new Amount(sellAmount)
             .multiplyByDecimals(sellAsset?.decimals ?? 18)
```
Comment (supermassive) [inline components/brave_wallet_ui/common/hooks/use-multi-chain-sell-assets.ts:81]: There should be no fallback to 18 as for BTC it is 8. 
Actually there should be no fallback at all as sellAsset is checked above to be truthy

**Review by supermassive — APPROVED** [review]
wallet core lgtm

File: components/brave_wallet_ui/common/hooks/use-multi-chain-sell-assets.ts (hunk 2 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/common/hooks/use-multi-chain-sell-assets.ts]: Removed the fallback!

**Review by onyb — APPROVED** [review]
++

**Comment by srirambv** [comment]
Verification passed on 

Brave | 1.64.61 Chromium: 121.0.6167.164 (Official Build) nightly (64-bit)
-- | --
Revision | 905da4d4305a63a9dcf5717062022e8a6216f688
OS | Linux

- Verified steps from PR description
- Verified BTC Off-Ramp works as expected

https://github.com/brave/brave-core/assets/17010094/3cfaf4da-ed4c-4124-8bb8-bc6d26952206

**Pull request merged**